junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertThrowsExactly.java

```diff
@@ -60,15 +58,19 @@ private static <T extends Throwable> T assertThrowsExactly(Class<T> expectedType
 			}
 			else {
 				UnrecoverableExceptions.rethrowIfUnrecoverable(actualException);
-				String message = buildPrefix(nullSafeGet(messageOrSupplier))
-						+ format(expectedType, actualException.getClass(), "Unexpected exception type thrown");
-				throw new AssertionFailedError(message, actualException);
+				throw assertionFailure() //
+						.message(messageOrSupplier) //
+						.expected(expectedType) //
+						.actual(actualException.getClass()) //
+						.reason("Unexpected exception type thrown") //
+						.build();
```

Comment:
Shouldn't `actualException` be handed over using `cause(actualException)` or was it intended to omit it here?

Reply:
Good catch! Fixed in 1e7da1152253a289c963c1d234f3494ffe172d34.